OpenRA.Mods.Common/Traits/Player/PlaceBuilding.cs

```diff
@@ -135,27 +135,26 @@ void IResolveOrder.ResolveOrder(Actor self, Order order)
 				}
 				else if (os == "PlacePlug")
 				{
-					var host = self.World.WorldActor.Trait<BuildingInfluence>().GetBuildingAt(targetLocation);
-					if (host == null)
-						return;
-
 					var plugInfo = actorInfo.TraitInfoOrDefault<PlugInfo>();
 					if (plugInfo == null)
 						return;
 
-					var location = host.Location;
-					var pluggableLocations = host.TraitsImplementing<Pluggable>()
-						.Where(p => p.AcceptsPlug(host, plugInfo.Type));
+					foreach (var a in self.World.ActorMap.GetActorsAt(targetLocation))
+					{
+						var pluggables = a.TraitsImplementing<Pluggable>()
+							.Where(p => p.AcceptsPlug(a, plugInfo.Type))
+							.ToList();
 
-					var pluggable = pluggableLocations.FirstOrDefault(p => location + p.Info.Offset == targetLocation)
-						?? pluggableLocations.FirstOrDefault();
+						var pluggable = pluggables.FirstOrDefault(p => a.Location + p.Info.Offset == targetLocation) ??
+						                pluggables.FirstOrDefault();
```

Comment:
Why do it in two steps?

Reply:
Maybe i'm missing something - how would I do this in one?

Reply:
I looked to quick!